In [1]:
import pandas as pd
import csv

In [2]:
grants_1990s = pd.read_csv('https://raw.githubusercontent.com/ejt2703/neh_grant_data_gender/main/NEH_Grants1990s.csv')

In [3]:
grants_2000s = pd.read_csv('https://raw.githubusercontent.com/ejt2703/neh_grant_data_gender/refs/heads/main/NEH_Grants2000s.csv')

In [4]:
grants_2010s = pd.read_csv('https://raw.githubusercontent.com/ejt2703/neh_grant_data_gender/refs/heads/main/NEH_Grants2010s.csv')

In [5]:
grants_2020s = pd.read_csv('https://raw.githubusercontent.com/ejt2703/neh_grant_data_gender/refs/heads/main/NEH_Grants2020s.csv')

In [6]:
# AppNumber,ApplicantType,Institution,OrganizationType,InstCity,InstState,InstPostalCode,InstCountry,CongressionalDistrict,Latitude,Longitude,CouncilDate,YearAwarded,ProjectTitle,Program,Division,ApprovedOutright,ApprovedMatching,AwardOutright,AwardMatching,OriginalAmount,SupplementAmount,BeginGrant,EndGrant,ProjectDesc,ToSupport,PrimaryDiscipline,SupplementCount,Supplements,ParticipantCount,Participants,DisciplineCount,Disciplines

In [7]:
combined_grants = [grants_1990s, grants_2000s, grants_2010s, grants_2020s]
combined_df = pd.concat(combined_grants, ignore_index=True)

In [8]:
#combined_df.to_csv("combined_df.csv", index=False)

In [9]:
gender_keywords = [
'GENDER',
 'GENDERED',
 'GENDERING',
 'SEXUALITY',
 'SEXUALIZE',
 'SEXUALIZATION',
 'SEXUAL',
 'LGBTQ+',
 'LGBT',
 'LGBT+',
 'LGBTQIA+',
 'LGBTQIA',
 'LESBIAN',
 'GAY',
 'BISEXUAL',
 'ASEXUAL',
 'TRANSGENDER',
 'QUEER',
 'QUEERED',
 'QUEERING',
 'WOMAN',
 'WOMEN',
 'WOMANHOOD',
 'FEMALE',
 'FEMALES',
 'GIRL',
 'GIRLS',
 'GIRLHOOD',
 'FEMININE',
 'FEMININITY',
 'MASCULINE',
 'MASCULINITY',
 'FEMINISM',
 'FEMINIST',
 'FEMINISMS',
 'PATRIARCHY',
 'PATRIARCHAL',
 'HETERONORMATIVE',
 'HETERONORMATIVITY',
 'CISGENDER',
 'CISNORMATIVE',
 'CISNORMATIVITY',
 'EMILY DICKINSON',
 'FRANCES PERKINS',
 'SHIRLEY CHISHOLM',
 'MARY SHELLEY',
 'CATHARINE',
 'SOR JUANA',
 'ESTHER',
 'HANNAH',
 'LOUISA',
 'ISADORA',
 'MME DE STAEL',
 'OCTAVIA BUTLER',
 'AGNES',
 'GABRIELA MISTRAL',
 'ISABEL',
 'LUISA',
 'UNA JEFFERS',
 'HATSHEPSUT',
 'MINÉ OKUBO',
 'BERNARDINE',
 'GLBT',
 'GODDESS',
 'ELIZABETH',
 'CLEOPATRA',
 'CHARLOTTE',
 'BRONTE',
 'WOOLF',
 'TONI MORRISON',
 'MAYA ANGELOU',
 'BELL HOOKS',
 'ALICE WALKER',
 'ANNA JULIA COOPER',
 'ZORA NEALE HURSTON',
 'FRIDA KAHLO',
 'CELIA CRUZ',
 'RIGOBERTA MENCHÚ',
 'RIGOBERTA MENCHU',
 'SANDRA CISNEROS',
 'GLORIA ANZALDÚA',
 'GLORIA ANZALDUA',
 'JULIA DE BURGOS',
 'ESMERALDA SANTIAGO',
 'LORNA DEE CERVANTES',
 'LORRAINE HANSBERRY',
 'ASSATA SHAKUR',
 'CHIMAMANDA NGOZI ADICHIE',
 'LESLIE MARMON SILKO',
 'LOUISE ERDRICH',
 'JOY HARJO',
 'LINDA HOGAN',
 'ROBIN WALL KIMMERER',
 'WINONA LADUKE',
 'AUNG SAN SUU KYI',
 'YOKO ONO',
 'MAXINE HONG KINGSTON',
 'AMY TAN',
 'CATHY PARK HONG',
 'OCEAN VUONG',
 'THERESA HAK KYUNG CHA',
 'GAYATRI SPIVAK',
 'GHADA AL-SAMMAN',
 'ASSIA DJEBAR',
 'LEILA ABOULELA',
 'MARJANE SATRAPI',
 'FATEMA MERNISSI',
 'SAPPHO',
 'SAPPHIC',
 'SAPPHISM',
 'HAUNANI-KAY TRASK',
 'TERESIA TEAIWA',
 'SELINA TUSITALA MARSH',
 'PATRICIA GRACE',
 'GRACE MERA MOLISA',
 'SUSAN',
 'DAUGHTER',
 'ADELAIDE JOHNSON',
 'ALICE CHENOWETH',
 'HELEN HAMILTON GARDENER,ALMA THOMAS',
 'AMY LOWELL',
 'ANNA JULIA COOPER',
 'BEATRIX POTTER',
 'BELLE DA COSTA GREENE',
 'CHARLOTTE PERKINS GILMAN',
 'EUDORA WELTY',
 'ESMERALDA SANTIAGO',
 'HANNAH CRAFTS',
 'ISABELLA STEWART GARDNER',
 'JANE ADDAMS',
 'JOAN OF ARC',
 'JULIA ALVAREZ',
 'JULIA ÁLVAREZ',
 'LOIE FULLER',
 'LOÏE FULLER',
 'KATHARINE',
 'LA MALINCHE',
 'MFK FISHER',
 'M.F.K. FISHER',
 'MARTHA MOORE BALLARD',
 'MARY CASSATT',
 'MOTHER CABRINI',
 'MOTHER FRANCES XAVIER CABRINI',
 'MOTHER THERESA',
 'NATALIE DE BLOIS',
 'RUTH BADER GINSBURG',
 'SYLVIA PLATH',
 'URSULA PARROTT',
 'URSULA LE GUIN',
 'URSULA K. LE GUIN',
 'URSULA K LE GUIN',
 'WINONA LADUKE'
 ]

In [10]:
gender_discipline = [
    "GENDER STUDIES",
    "WOMEN'S HISTORY"
]

In [11]:
info_columns = ['ProjectTitle', 'ProjectDesc', 'ToSupport']
discipline_columns = ['PrimaryDiscipline', 'Disciplines']

In [12]:
# Below is my original code to establish a gender_filter.
# It didn't work because pandas needed to evaluate it as a string, which is was not.

# def gender_filter(row):
#    for column in info_columns:
#        if any(keyword in row[column].upper() for keyword in gender_keywords):
#            return True
#    for column in discipline_columns:
#        if any(keyword in row[column].upper() for keyword in gender_discipline):
#            return True
#    return False

# Below, the code is revised to convert each row[column] to a string before applying the upper() method.

def gender_filter(row):
    for column in info_columns:
        if any(keyword in str(row[column]).upper() for keyword in gender_keywords):
            return True
    for column in discipline_columns:
        if any(keyword in str(row[column]).upper() for keyword in gender_discipline):
            return True
    return False

In [13]:
combined_df['gender_focus'] = combined_df.apply(gender_filter, axis=1)

# apply lets me run the function across rows or columns
# axis=1 tells pandas to specifically apply the function to each row of the dataframe

In [14]:
combined_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 34632 entries, 0 to 34631
Data columns (total 34 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   AppNumber              34632 non-null  object 
 1   ApplicantType          34632 non-null  int64  
 2   Institution            34632 non-null  object 
 3   OrganizationType       34632 non-null  object 
 4   InstCity               34632 non-null  object 
 5   InstState              34632 non-null  object 
 6   InstPostalCode         34628 non-null  object 
 7   InstCountry            34632 non-null  object 
 8   CongressionalDistrict  34632 non-null  int64  
 9   Latitude               34632 non-null  object 
 10  Longitude              34632 non-null  object 
 11  CouncilDate            34632 non-null  object 
 12  YearAwarded            34632 non-null  int64  
 13  ProjectTitle           34632 non-null  object 
 14  Program                34632 non-null  object 
 15  Di

In [15]:
combined_df['YearAwarded'].dtype

dtype('int64')

In [16]:
# This line removes all rows where the YearAwarded is before 1990.

combined_df = combined_df.drop(
    combined_df[ combined_df['YearAwarded'] < 1990 ].index
)

In [22]:
len(combined_df)

34360

Now, combined_df only has projects awarded in and after 1990, AND it has a gender_focus column, in which I can assess projects that have a gender focus (TRUE) or not (FALSE) based on a list of gender_keywords and gender_disciplines.

Now I want to answer the following quantifiable questions:
1. How many funded projects between 1990 and 2025 had a gender focus? What percentage?
2. How has the number of gender-related projects changed over the years?
3. What states have conducted the most gender-related projects?
4. How much funding in Outright Awards have gender-related projects received between 1990 and 2025?
5. About how much funding goes to gender-related projects each year? (median or mean)

In [ ]:
# This line tells me how many projects with and without a gender_focus are in the dataframe.

combined_df['gender_focus'].value_counts()

gender_focus
False    31139
True      3221
Name: count, dtype: int64

*Between 1990 and 2025, 3,221 out of 34,360 projects had something to do with gender.*

In [ ]:
# This line creates a table showing projects with/without a gender_focus by year.

gender_focus_by_year = combined_df.groupby(["YearAwarded", "gender_focus"]).size().rename("projects").reset_index()

In [37]:
# Using "pivot", this code creates a pivot table by identifing the index (row) as gender_focus,
# the columns as YearAwarded,
# and the values inside them as the projects.

# "fillna" means that any empty cells without a number (NaN - "not a number")
# will be filled with 0 instead of NaN.
# So if there are no gender-related projects in a year, Python will put a 0 instead of NaN.

gender_year_pivot_table = gender_focus_by_year.pivot(
    index="YearAwarded",
    columns="gender_focus",
    values="projects"
).fillna(0)

In [ ]:
gender_year_pivot_table

# This could be good to visualize in a stacked bar chart
# with a line graph overlay, showing the rate of change.

gender_focus  False  True 
YearAwarded               
1990           1820    114
1991           1613    108
1992           1757    125
1993           1519    117
1994           1369    154
1995           1263    125
1996            478     33
1997            594     57
1998            593     36
1999            709     52
2000            972     55
2001           1066     59
2002           1011     61
2003            853    144
2004            949     82
2005            868     70
2006            907     56
2007            741     65
2008            677     81
2009            816     86
2010            902    121
2011            633     76
2012            660     79
2013            552     77
2014            503    104
2015            673     86
2016            704    111
2017            649     96
2018            612     83
2019            556    103
2020            904    116
2021            949    128
2022            645    115
2023            804    124
2024            657    109
2025            161     13

In [ ]:
# This line creates a table showing projects with/without a gender_focus by state.

gender_focus_by_state = combined_df.groupby(["InstState", "gender_focus"]).size().rename("projects").reset_index()

In [ ]:
gender_state_pivot_table = gender_focus_by_state.pivot(
    index="InstState",
    columns="gender_focus",
    values="projects"
).fillna(0)

In [ ]:
gender_state_pivot_table

# This could be good to visualize in a choropleth map of the U.S.
# Perhaps compare with electoral map? Are there more gender projects in blue states?

gender_focus   False  True 
InstState                  
AK             165.0    0.0
AL             244.0   23.0
AR             155.0   15.0
AS              22.0    0.0
AZ             391.0   24.0
CA            2578.0  295.0
CO             371.0   35.0
CT             661.0   62.0
DC             781.0   64.0
DE             167.0   12.0
FL             636.0   70.0
GA             546.0   62.0
GU              33.0    0.0
HI             193.0   12.0
IA             334.0   33.0
ID             122.0    9.0
IL            1492.0  138.0
IN             723.0   60.0
KS             269.0   43.0
KY             287.0   28.0
LA             374.0   29.0
MA            2135.0  253.0
MD             756.0   79.0
ME             303.0   19.0
MI             767.0   76.0
MN             563.0   60.0
MO             453.0   48.0
MP              27.0    0.0
MS             246.0   19.0
MT             174.0    6.0
NC             879.0  145.0
ND              89.0    6.0
NE             199.0   17.0
NH             290.0   17.0
NJ             710.0   84.0
NM             287.0   19.0
NV             104.0    8.0
NY            3989.0  461.0
OH             869.0  108.0
OK             227.0   17.0
OR             426.0   43.0
PA            1529.0  171.0
PR             131.0    5.0
RI             297.0   30.0
SC             307.0   28.0
SD             105.0    3.0
TN             441.0   44.0
TX            1197.0  148.0
UT             200.0   17.0
Un             191.0   13.0
VA            1191.0  138.0
VI              34.0    1.0
VT             252.0   17.0
WA             445.0   47.0
WI             547.0   48.0
WV             100.0    5.0
WY             135.0    7.0

In [58]:
gender_focus_by_award = combined_df.groupby("gender_focus")["AwardOutright"].sum()

In [59]:
gender_focus_by_award

gender_focus
False   3,786,330,139.50
True      195,929,273.09
Name: AwardOutright, dtype: float64

In [ ]:
3786330139.50 + 195929273.09

# Total Outright Awards: $3,978,562,412.59

3982259412.59

*Between 1990 and 2025, gender-related projects have received 195 million dollars in funding, compared to almost 3.7 billion dollars in funding for non- gender-related projects.*

*The total outright awards are $3,978,562,412.59.*

In [ ]:
# aggfunc means "aggregation function", which tells pandas how to summarize the data.
# Here, I use "sum" to add up all the AwardOutright amounts for each year for projects with and without a gender_focus.
# fill_value=0, like fillna from before, ensures that any empty cells are filled with 0 instead of NaN.

award_per_year_pivot = combined_df.pivot_table(
    index="YearAwarded",
    columns="gender_focus",
    values="AwardOutright",
    aggfunc="sum",
    fill_value=0
)

In [ ]:
award_per_year_pivot

# This could be good to visualize in a stacked bar chart.
# Or, the funding amounts could be shown as the line graph overlaying my stacked bar chart for gender_year_pivot_table.

gender_focus          False         True 
YearAwarded                              
1990         112,781,858.11  1,788,001.57
1991         147,195,401.30  2,340,885.81
1992         127,498,711.15  2,002,201.78
1993         102,426,363.51  4,347,493.83
1994         112,084,848.61  4,451,225.22
1995          99,317,536.67  4,968,236.84
1996          93,593,491.47  1,784,487.66
1997          80,920,180.59  2,689,874.37
1998          94,918,287.35  1,097,424.00
1999          49,665,131.73  2,582,285.71
2000          84,204,720.09  2,585,171.31
2001         106,036,246.02  3,648,907.00
2002          61,370,133.00  3,559,850.00
2003          96,961,119.00  3,897,209.00
2004         122,778,220.68  3,426,729.00
2005          68,001,640.80  4,687,100.00
2006         101,000,539.84  2,773,425.00
2007         132,274,245.74  4,774,655.00
2008          67,905,555.63  4,502,945.50
2009         115,392,394.93  8,466,395.28
2010         148,609,173.99  6,922,900.27
2011          76,227,180.60  5,631,856.70
2012         101,883,663.61  6,945,848.23
2013         127,868,449.95  4,793,795.41
2014          58,667,437.17  8,196,729.46
2015         110,306,663.74  5,988,426.79
2016         138,307,411.93  8,037,733.00
2017          63,724,618.05  5,717,793.27
2018         113,784,812.19  8,425,175.81
2019         164,691,377.92  7,706,757.73
2020         108,524,272.52 10,869,432.05
2021         253,809,519.05 16,296,648.45
2022         161,826,324.66  8,810,859.40
2023          86,889,445.90 10,027,954.64
2024          84,552,205.00 10,074,881.00
2025          10,330,957.00  1,107,977.00

Next, I want to compare overall trends in the NEH's gender priorities with the recent funding cuts.

1. Of the projects that had their funding cut in April 2025, how many had a gender_focus? What percentage?
2. What were the names of those projects?
3. How much funding did the projects whose grants were cancelled receive? How does that number compare to the average or median amount of funding that has been awarded to gender-related projects per year?

Lastly, I want to take a look at the new grants that were issued in August. The NYT reported that many of these grants promote American ideals and focus on presidents, statesmen, and canonical figures in American history.

1. Of the new grants that were issued, how many supported projects with a gender_focus? What percentage?
2. How much funding did gender_related projects receive?
3. What primary disciplines did each new project belong too? What is the most popular discipline?